src/control/server/server.go

```diff
@@ -263,29 +289,43 @@ func Start(log *logging.LeveledLogger, cfg *config.Server) error {
 				return err
 			}
 
+			if !sysdb.IsReplica() {
+				continue
+			}
+
 			// Start the system db after instance 0's SCM is
 			// ready.
 			var once sync.Once
 			srv.OnStorageReady(func(ctx context.Context) (err error) {
 				once.Do(func() {
-					err = errors.Wrap(sysdb.Start(ctx, controlAddr),
+					err = errors.Wrap(sysdb.Start(ctx),
 						"failed to start system db",
 					)
 				})
 				return
 			})
+
+			if !sysdb.IsBootstrap() {
```

Comment:
this check and updating srv.joinSystem could be done before NewIOServerInstance and instead of reassigning the closure, just pass the bootstrap flag into the constructor to handle the assignment of the closure